src/test/java/ai/elimu/rest/v2/crowdsource/LetterToAllophoneMappingsControllerTest.java

```diff
@@ -0,0 +1,30 @@
+package ai.elimu.rest.v2.crowdsource;
+
+import ai.elimu.util.JsonLoader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+import selenium.DomainHelper;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.assertThat;
+
+public class LetterToAllophoneMappingsControllerTest {
+    private Logger logger = LogManager.getLogger();
+
+    @Test
+    public void testGetWordDataForCrowdSourcing() {
+        String jsonResponse = JsonLoader.loadJson(DomainHelper.getRestUrlV2() + "/crowdsource/word-contributions/letter-to-allophone-mappings");
+        logger.info("word-dataResponse: " + jsonResponse);
+
+        JSONArray allophoneJSONArray = new JSONArray(jsonResponse);
+        logger.info("wordsJSONArray.length(): " + allophoneJSONArray.length());
+        assertThat(allophoneJSONArray.length() > 0, is(true));
+
+        JSONObject allophoneJsonObject = allophoneJSONArray.getJSONObject(0);
+        assertThat(allophoneJsonObject.getLong("id"), not(nullValue()));
```

Comment:
This test might need some refactoring because the response does not contain an array of Letters and an array of Allophones. Instead, the response consists of an array of LetterToAllophoneMappings, and within each mapping there is an array of Letters and an array of Allophones: https://github.com/elimu-ai/model/blob/master/src/main/java/ai/elimu/model/v2/gson/content/LetterToAllophoneMappingGson.java

But this will be easier to test after we actually deploy the code to the test server, of course.

Reply:
@nya-elimuai yes this is a mistake. will refactor it.